integrations/openzeppelin/contracts.md

```diff
@@ -0,0 +1,235 @@
+ - **Ownable.sol** — extension to restrict access to certain functions
+
+The mintable ERC20 OpenZeppelin token contract provides a `mint` function that the owner of the contract can only call. By default, the owner is the contract's deployer address. There is also a premint of `1000` tokens sent to the contract's deployer configured in the `constructor` function.
+
+The first step is to go to [Remix](https://remix.ethereum.org/) and take the following steps:
+
+ 1. Click on the "Create New File" icon and set a file name. For this example, it was set to `ERC20.sol`
+ 2. Make sure the file was created successfully. Click on the file to open it up in the text editor
+ 3. Write your smart contract using the file editor. For this example, the following code was used:
+
+```sol
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MyToken is ERC20, Ownable {
+    constructor() ERC20("MyToken", "MTK") {
+        _mint(msg.sender, 1000 * 10 ** decimals());
+    }
+
+    function mint(address to, uint256 amount) public onlyOwner {
+        _mint(to, amount);
+    }
+}
+```
+
+![Getting Started with Remix](/images/openzeppelin/ozcontracts-images1.png)
+
+Once your smart contract is written, you can compile it by taking the following steps:
+
+ 1. Head to the "Solidity Compiler`
+ 2. Click on the compile button
+ 3. Alternatively, you can check the "Auto compile" feature
+
+![Compile ERC20 Contract with Remix](/images/openzeppelin/ozcontracts-images2.png)
+
+With the contract compiled, you are ready to deploy it taking the following steps: 
+
+ 1. Head to the "Deploy & Run Transactions" tab
+ 2. Change the environment to "Injected Web3". This will use MetaMask's injected provider. Consequently, the contract will be deployed to whatever network MetaMask is connected to. MetaMask might show a pop-up outlining that Remix is trying to connect to your wallet
+ 3. Select the proper contract to deploy. In this example, it is the `MyToken` contract inside the `ERC20.sol` file
+ 4. If everything is ready, click on the "Deploy" button. Review the transaction information in MetaMask and confirm it
+ 5. After a few seconds, the transaction should get confirmed, and you should see your contract under "Deployed Contracts"
+
+![Deploy ERC20 Contract with Remix](/images/openzeppelin/ozcontracts-images3.png)
+
+And that is it! You've deployed an ERC20 token contract using OpenZeppeling's contracts and libraries. Next, you can interact with your token contract via Remix, or add it to MetaMask.
+
+### Deploying an ERC721 Token
+
+For this example, an ERC721 token will be deployed to Moonbase Alpha. The final code used combines different contracts from OpenZeppelin:
+
+ - **ERC721** — ERC721 token implementation with the optional features from the base interface. Includes the supply mechanism with a `_mint` function but needs to be explicitly called from within the main contract
```

Comment:
commented on this with the ERC20 example, but I am confused about what are "the optional features" and what is "the supply mechanism"?

Reply:
I hope my previous answer explained a little bit better the supply mechanism, this was mostly extracted from their site as well.